### PR TITLE
LRDOCS-4111 JSdoc Ant task

### PR DESCRIPTION
**Pull request by mwilliams2014**
@jbalsas Just install jsdoc and run `ant jsdoc` from the liferay-portal root folder to generate the docs. They'll be in the `api\jsdoc` folder. I've also added extra jsdoc to `ToggleDisableInputs.es.js` to show how we can have JSdoc for STATE properties. Let me know if that's something we want to do. Please let me know if this is covering the proper modules. Thanks!

### CODE REVIEW

File: modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js [hunk 1]
```diff
@@ -72,6 +78,8 @@ ToggleDisableInputs.STATE = {
 	/**
 	 * Label of the switch
 	 * @type {String}
+	 * @memberof! ToggleDisableInputs#
```
Comment (jbalsas) [inline modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: Why the additional exclamation mark? This should be just `@memberof`

File: modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js [hunk 2]
```diff
@@ -72,6 +78,8 @@ ToggleDisableInputs.STATE = {
...
+	 * @instance ToggleDisableInputs.STATE.label
```
Comment (jbalsas) [inline modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: Why the extra description after `@instance`? Just the annotation should be enough

Comment (mwilliams2014) [reply modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: Same as above. I saw it done that way in an example doc

File: modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js (hunk 1 above)
Comment (mwilliams2014) [reply modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: That was the syntax I saw in an example

**Comment by mwilliams2014** [comment]
@jbalsas I'm not sure why it's failing for you. Perhaps it's your local environment. I tested on Windows, so I'm not sure. Did you npm install jsdoc ? 

I'll have to have someone else on a macbook test to see if they get the same results. I'll make the updates you suggested and take it from there. Thanks!

File: modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js (hunk 1 above)
Comment (jbalsas) [reply modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: We're using it this way: https://github.com/liferay/alloy-editor/search?utf8=%E2%9C%93&q=memberof&type=

The exclamation mark does the following:

> The "forced" @memberof tag, @memberof!, forces the object to be documented as belonging to a specific parent even if it appears to have a different parent.

Unless we need it, better to stay off that path 😉

**Comment by jbalsas** [comment]
I did install jsdoc globally through npm. It started things and just failed. No other logs showed up.

File: modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js (hunk 1 above)
Comment (mwilliams2014) [reply modules/apps/web-experience/portlet-configuration/portlet-configuration-css-web/src/main/resources/META-INF/resources/js/ToggleDisableInputs.es.js]: Ah. Will do ;-)

**Comment by mwilliams2014** [comment]
K. Thanks! I'll have to keep at it then and debug it.

**Comment by mwilliams2014** [comment]
I had a coworker test on Mac and everything worked. I'm not sure why you ran into that issue. I'm going to push forward with this and make the changes you suggested. Thanks!

**Comment by mwilliams2014** [comment]
Hey @jbalsas do you know how to increase JavaScript heap memory size. Someone else ran into this issue as well. It's odd because it happens for some and not for others...

**Comment by jbalsas** [comment]
https://github.com/jsdoc3/jsdoc/issues/1368

**Comment by mwilliams2014** [comment]
Thanks @jbalsas ! I'll see if removing some of the `@module` tags works. Otherwise, yeah, I guess it is a bug with JSDoc itself...

**Pull request closed without merging**